editor/filesystem_dock.cpp

```diff
@@ -3861,8 +3861,20 @@ void FileSystemDock::load_layout_from_config(Ref<ConfigFile> p_layout, const Str
 
 	if (p_layout->has_section_key(p_section, "dock_filesystem_selected_paths")) {
 		PackedStringArray dock_filesystem_selected_paths = p_layout->get_value(p_section, "dock_filesystem_selected_paths");
+		PackedStringArray dock_filesystem_really_selected_paths;
+		bool dock_filesystem_selected_paths_changed = false;
+		Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
 		for (int i = 0; i < dock_filesystem_selected_paths.size(); i++) {
-			select_file(dock_filesystem_selected_paths[i]);
+			auto p_path = dock_filesystem_selected_paths[i];
+			if (da->file_exists(p_path) || da->dir_exists(p_path)) {
```

Comment:
Stored directories end with `/`, so you can use that fact instead of checking both for file and directory.